File: frontend/src/types/Dish.ts
export interface IDish {
  id: number;
  name: string;
  time_to_cook: number;
  recipe: string;
  description: string;
  category_id: number;
  created_at: Date;
}

export interface IDishWithImages extends IDish {
  images: string[];
}

export interface ICreateDish {
  name: string;
  time_to_cook: number;
  recipe: string;
  description: string;
  category_id: number;
}

File: backend/internal/dto/dish.go
package dto

import "time"

type CreateDishDTO struct {
	Name        string `json:"name"`
	TimeToCook  int    `json:"time_to_cook"`
	Recipe      string `json:"recipe"`
	Description string `json:"description"`
	CategoryID  int    `json:"category_id"`
}

type UpdateDishDTO struct {
	Name        string `json:"name"`
	TimeToCook  int    `json:"time_to_cook"`
	Recipe      string `json:"recipe"`
	Description string `json:"description"`
}

type DishDTO struct {
	ID          int       `json:"id"`
	Name        string    `json:"name"`
	TimeToCook  int       `json:"time_to_cook"`
	Recipe      string    `json:"recipe"`
	Description string    `json:"description"`
	CategoryID  int       `json:"category_id"`
	Image       *string   `json:"image"`
	CreatedAt   time.Time `json:"created_at"`
}

type GetDishDTO struct {
	DishDTO
	Images []string `json:"images"`
}
